src/org/sipdroid/sipua/SipdroidEngine.java

```diff
@@ -593,6 +620,7 @@ public void onMWIUpdate(RegisterAgent mwi_ra, boolean voicemail, int number, Str
 			Receiver.onText(Receiver.MWI_NOTIFICATION, null, 0,0);
 			lastmsgs[i] = null;
 		}
+		unPublishAll();

```

Comment:
why did you put ```unPublishAll()``` in the ```onMWIUpdate``` function?  Is this something you put in the wrong place after creating the new branch?  Can you check in your patch file?